ykrt/src/mt.rs

```diff
@@ -407,50 +438,69 @@ mod tests {
         let mut mtt = MTBuilder::new().hot_threshold(2).init();
 
         // The program is silly. Do nothing twice, then start again.
-        enum ByteCode {
-            Nop,
-            Restart,
-        }
-        let prog = vec![ByteCode::Nop, ByteCode::Nop, ByteCode::Restart];
+        const NOP: u8 = 0;
+        const RESTART: u8 = 1;
+        let prog = vec![NOP, NOP, RESTART];
 
         // Suppose the bytecode compiler for this imaginary language knows that the first bytecode
         // is the only place a loop can start.
         let locs = vec![Some(Location::new()), None, None];
 
         struct IO {
-            prog: Vec<ByteCode>,
+            prog: Vec<u8>,
             pc: usize,
+            count: usize,
         }
 
-        let interp_step = |mut tio: IO| {
+        #[interp_step]
+        fn dumb_interp_step(tio: &mut IO) {
+            // FIXME make `inputs` a struct. Named fields would be much nicer.
             match tio.prog[tio.pc] {
-                ByteCode::Nop => tio.pc += 1,
-                ByteCode::Restart => tio.pc = 0,
+                NOP => {
+                    tio.pc += 1;
+                    tio.count += 1;
+                }
+                RESTART => tio.pc = 0,
+                _ => unreachable!(),
             }
-            tio
-        };
+        }
 
-        let mut tio = IO { prog, pc: 0 }; // bytecode, program counter.
+        let mut tio = IO {
+            prog,
+            pc: 0,
+            count: 0,
+        }; // bytecode, program counter.
```

Comment:
I think that comment is a leftover from when tio was a tuple also. Let's kill it?

Reply:
Fixed in 5638b7f.